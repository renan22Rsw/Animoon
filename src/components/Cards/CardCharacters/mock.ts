interface mockTypes {
  id: number;
  name: string;
  image: string;
  role: string;
  voices: {
    id: number;
    name: string;
    image: string;
    language: string;
  }[];
}

export const mock: mockTypes = {
  id: 1,
  name: "Naruto",
  image: "https://example.com/character-image.jpg",
  role: "Main",
  voices: [
    {
      id: 1,
      name: "Junko Takeuch",
      image: "https://example.com/voices-image.jpg",
      language: "Japanease",
    },
  ],
};
