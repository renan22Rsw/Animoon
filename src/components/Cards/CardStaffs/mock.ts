interface mockTypes {
  id: number;
  name: string;
  image: string;
  occupation: string;
}

export const mock: mockTypes = {
  id: 1,
  name: "Masashi Kishimoto",
  image: "https://example.com/staff-image.jpg",
  occupation: "Original Creator",
};
