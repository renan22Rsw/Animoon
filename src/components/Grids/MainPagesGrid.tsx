import React from "react";
import MainCard from "../Cards/MainCard";

interface MainPagesGridProps {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface CharacterMainPage {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

interface MainPagesGridData {
  datas: (MainPagesGridProps | CharacterMainPage)[];
}

const MainPagesGrid = ({ datas }: MainPagesGridData) => {
  return (
    <div className="grid grid-cols-3 text-center md:grid-cols-3 lg:grid-cols-6 md:mx-10">
      {datas.map((data) => {
        const title = "title" in data ? data.title.romaji : data.name.full;
        const image =
          "coverImage" in data ? data.coverImage.large : data.image.large;

        return (
          <MainCard key={data.id} title={title} images={image} id={data.id} />
        );
      })}
    </div>
  );
};

export default MainPagesGrid;
