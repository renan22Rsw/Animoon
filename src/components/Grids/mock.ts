interface mockAnimesAndMangas {
  id: number;
  title: {
    romaji: string;
  };
  coverImage: {
    large: string;
  };
}

interface mockCharacter {
  id: number;
  name: {
    full: string;
  };
  image: {
    large: string;
  };
}

export const mockAnimesAndMangasMainPage: mockAnimesAndMangas = {
  id: 1,
  title: { romaji: "Akame ga kill" },
  coverImage: { large: "https://example.com/anime-image.jpg" },
};

export const mockCharacterMainPage: mockCharacter = {
  id: 2,
  name: { full: "Akame" },
  image: { large: "https://example.com/character-image.jpg" },
};
