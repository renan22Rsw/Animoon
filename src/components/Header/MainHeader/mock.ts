interface mockTypes {
  title: string;
  description: string;
  coverImage: string;
}

export const mock: mockTypes = {
  title: "Naruto",
  description:
    "Naruto Uzumaki, a hyperactive and knuckle-headed ninja, lives in Konohagakure, the Hidden Leaf village. Moments prior to his birth, a huge demon known as the Kyuubi, the Nine-tailed Fox, attacked Konohagakure and wreaked havoc. In order to put an end to the Kyuubi's rampage, the leader of the village, the 4th Hokage, sacrificed his life and sealed the monstrous beast inside the newborn Naruto.",
  coverImage: "https://example.com/image.jpg",
};
