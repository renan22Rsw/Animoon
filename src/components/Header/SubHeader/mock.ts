interface mockTypes {
  name: string;
  nativeName: string;
  alternativeName?: string[];
  age: string;
  image: string;
  gender: string;
  month: number;
  day: number;
  bloodType: string;
  hometown: string;
  yearsActive?: number[];
  description: string;
}

export const mock: mockTypes = {
  name: "Naruto",
  nativeName: "Naruto Uzumaki",
  alternativeName: ["Nine-Tails Jinchuuriki", " Number One Unpredictable"],
  age: "17",
  image: "http://example.com/image.jpg",
  gender: "Male",
  month: 10,
  day: 10,
  bloodType: "B",
  hometown: "Saitama Prefecture, Japan",
  yearsActive: [1996],
  description: "Born in Konohagakure, a ninja village hidden in the leaves",
};
