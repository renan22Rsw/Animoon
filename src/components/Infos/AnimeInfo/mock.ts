interface mockTypes {
  format: string;
  duration: number;
  status: string;
  seasonYear: number;
  season: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  duration: 24,
  averageScore: 85,
  meanScore: 88,
  favourites: 500,
  format: "TV",
  genres: ["comedy", "horror"],
  popularity: 1000,
  season: "winter",
  seasonYear: 2024,
  source: "Light Novel",
  status: "Airing",
};
