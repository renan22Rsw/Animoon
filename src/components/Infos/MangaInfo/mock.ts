interface mockTypes {
  format: string;
  status: string;
  averageScore: number;
  meanScore: number;
  popularity: number;
  favourites: number;
  source: string;
  genres: string[];
}

export const mock: mockTypes = {
  averageScore: 80,
  meanScore: 85,
  favourites: 1200,
  format: "Manga",
  genres: ["action", "adventure"],
  popularity: 1000,
  source: "Original",
  status: "Releasing",
};
