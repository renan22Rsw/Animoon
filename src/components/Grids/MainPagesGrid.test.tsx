import { render, screen } from "@testing-library/react";
import MainPagesGrid from "./MainPagesGrid";
import { mockAnimesAndMangasMainPage, mockCharacterMainPage } from "./mock";

const datas = [mockAnimesAndMangasMainPage, mockCharacterMainPage];

describe("MainPagesGrid component", () => {
  it("should render anime/mangas titles and character names", () => {
    render(<MainPagesGrid datas={datas} />);
    expect(screen.getByText("Akame ga kill")).toBeInTheDocument();
    expect(screen.getByText("Akame")).toBeInTheDocument();
  });

  it("should render animes/mangas/characters images", () => {
    render(<MainPagesGrid datas={datas} />);

    const images = screen.getAllByRole("img");
    expect(images[0]).toBeInTheDocument();
    expect(images[1]).toBeInTheDocument();
  });

  it("should render animes/mangas/characters links", () => {
    render(<MainPagesGrid datas={datas} />);
    const links = screen.getAllByRole("link");
    expect(links[0]).toBeInTheDocument();
    expect(links[1]).toBeInTheDocument();
  });

  it("should match a snapshot", () => {
    const { container } = render(<MainPagesGrid datas={datas} />);
    expect(container).toMatchSnapshot();
  });
});
