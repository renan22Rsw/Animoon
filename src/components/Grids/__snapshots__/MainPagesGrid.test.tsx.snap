// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPagesGrid component should match a snapshot 1`] = `
<div>
  <div
    class="grid grid-cols-3 md:ml-14 md:grid-cols-3 lg:grid-cols-6 lg:text-center "
  >
    <div
      class=" py-2 px-1 "
    >
      <a
        href="/default section/1"
      >
        <img
          alt="card-picture"
          class=" rounded-md cursor-pointer h-[190px] lg:hover:scale-110  lg:duration-700 lg:ease-in-out lg:h-[265px]"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="300"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fanime-image.jpg&w=640&q=100"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fanime-image.jpg&w=384&q=100 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fanime-image.jpg&w=640&q=100 2x"
          style="color: transparent; width: 165px; object-fit: cover;"
          width="300"
        />
      </a>
      <h2
        class="py-4  text-xs md:text-sm   font-bold"
      >
        Akame ga kill
      </h2>
    </div>
    <div
      class=" py-2 px-1 "
    >
      <a
        href="/default section/2"
      >
        <img
          alt="card-picture"
          class=" rounded-md cursor-pointer h-[190px] lg:hover:scale-110  lg:duration-700 lg:ease-in-out lg:h-[265px]"
          data-nimg="1"
          decoding="async"
          fetchpriority="high"
          height="300"
          src="/_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=640&q=100"
          srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=384&q=100 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fcharacter-image.jpg&w=640&q=100 2x"
          style="color: transparent; width: 165px; object-fit: cover;"
          width="300"
        />
      </a>
      <h2
        class="py-4  text-xs md:text-sm   font-bold"
      >
        Akame
      </h2>
    </div>
  </div>
</div>
`;
